### ISSUE TITLE
Windows 10 add dll directory error

### ISSUE THREAD

**Issue by coalman321** [ros2/rpyutils#7]
Hi,
It seems the `add_dll_directories_from_env()` call will fail if '.' enters the path list, specifically on windows 10. The issue can be solved by converting the prefix path to an absolute path before calling `os.add_dll_directory(prefix_path)`. I have provided an example fix below. I am not 100% sure if this is the correct method though. 

This is the error that occurs without the patch:
```
Traceback (most recent call last):
  File "C:\dev\ros2_galactic\ros2-windows\lib\demo_nodes_py\listener-script.py", line 33, in <module>
    sys.exit(load_entry_point('demo-nodes-py==0.14.3', 'console_scripts', 'listener')())
  File "C:\dev\ros2_galactic\ros2-windows\lib\demo_nodes_py\listener-script.py", line 25, in importlib_load_entry_point
    return next(matches).load()
    module = import_module(match.group('module'))
  File "c:\python38\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\demo_nodes_py\topics\listener.py", line 16, in <module>
    from rclpy.node import Node
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\rclpy\node.py", line 43, in <module>
    from rclpy.client import Client
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\rclpy\client.py", line 22, in <module>
    from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\rclpy\impl\implementation_singleton.py", line 32, in <module>
    rclpy_implementation = import_c_library('_rclpy_pybind11', package)
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\rpyutils\import_c_library.py", line 38, in import_c_library
    with add_dll_directories_from_env('PATH'):
  File "c:\python38\lib\contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "C:\dev\ros2_galactic\ros2-windows\Lib\site-packages\rpyutils\add_dll_directories.py", line 50, in add_dll_directories_from_env
    dll_dir_handles.append(os.add_dll_directory(prefix_path))
  File "c:\python38\lib\os.py", line 1109, in add_dll_directory
    cookie = nt._add_dll_directory(path)
OSError: [WinError 87] The parameter is incorrect: '.'
```
Example fix that resolves the issue:
```
abs_path = os.path.abspath(prefix_path)
dll_dir_handles.append(os.add_dll_directory(abs_path))
```

**Comment by jacobperron** [comment]
@coalman321 Thanks for the bug report! I think your proposed fix makes sense. Would you be willing to contribute a pull request? Adding a test case [here](https://github.com/ros2/rpyutils/blob/master/test/rpyutils/test_add_dll_directories.py) would be nice too :)

**Comment by clalancette** [comment]
See https://github.com/ros2/rpyutils/pull/8 , where I did this change, added a test, and cleaned up the existing tests.

**Comment by clalancette** [comment]
This was fixed by #8, so closing.
